### PR TITLE
Programab get Predicates

### PR DESCRIPTION
**Pull request by supertick**
image

### CODE REVIEW

File: src/main/java/org/myrobotlab/service/ProgramAB.java [hunk 1]
```diff
@@ -1452,4 +1466,19 @@ public Utterance publishUtterance(Utterance utterance) {
...
+    return response.msg;
+  }
+
+  @Override
+  public String translate(String text, String fromLang, String toLang) {
+    // FIXME implement - same mapping that current inmoov has ? en-US etc ..
+    // although all of those are localizations
+    // not languages only the 1st part is "en" in en-US
+    // return getResponse(text)
+    return null;
```
Comment (kwatters) [inline src/main/java/org/myrobotlab/service/ProgramAB.java]: this should be implemented.

File: src/main/java/org/myrobotlab/service/ProgramAB.java [hunk 2]
```diff
@@ -1452,4 +1466,19 @@ public Utterance publishUtterance(Utterance utterance) {
     return utterance;
   }
 
+  @Override
+  public String translate(String text) {
+    Response response = getResponse(text);
```
Comment (kwatters) [inline src/main/java/org/myrobotlab/service/ProgramAB.java]: This should call the translate method with the max complexity args.

File: src/main/java/org/myrobotlab/service/ProgramAB.java [hunk 3]
```diff
@@ -608,6 +609,19 @@ public void reloadSession(String userName, String botName) throws IOException {
     }
   }
 
+  /**
+   * Get all current predicates names and their values
+   * for the current session
+   * @return
+   */
+  public Map<String, String> getPredicates() {
```
Comment (kwatters) [inline src/main/java/org/myrobotlab/service/ProgramAB.java:616]: there should be a getPredicates(session) method.. and this method should call that with the default session.

File: src/main/java/org/myrobotlab/service/interfaces/Translator.java [hunk 4]
```diff
@@ -1,8 +1,29 @@
...
+   * Translate the incoming text from incoming language to target language This
+   * method will rely on the internal state of the translation service to
+   * determine incoming and target language
+   * 
+   * @param text
+   * @return
+   */
+  public String
+
+      translate(String text);
```
Comment (kwatters) [inline src/main/java/org/myrobotlab/service/interfaces/Translator.java:15]: This interface should allow you to set the source/ destination language .. o/w the interface won't support changing which language you are translating to by default.

File: src/main/java/org/myrobotlab/service/ProgramAB.java (hunk 3 above)
Comment (supertick) [reply src/main/java/org/myrobotlab/service/ProgramAB.java:616]: agreed fixed

File: src/main/java/org/myrobotlab/service/interfaces/Translator.java (hunk 4 above)
Comment (supertick) [reply src/main/java/org/myrobotlab/service/interfaces/Translator.java:15]: but thats really setting the bot

File: src/main/java/org/myrobotlab/service/ProgramAB.java (hunk 1 above)
Comment (supertick) [reply src/main/java/org/myrobotlab/service/ProgramAB.java]: removed

File: src/main/java/org/myrobotlab/service/ProgramAB.java (hunk 2 above)
Comment (supertick) [reply src/main/java/org/myrobotlab/service/ProgramAB.java]: removed

File: src/main/java/org/myrobotlab/service/interfaces/Translator.java (hunk 4 above)
Comment (supertick) [reply src/main/java/org/myrobotlab/service/interfaces/Translator.java:15]: removed

**Pull request merged**